File: geekbang/algorithm-models/3.3-递归代码模板/recursion.js
// JavaScript
const recursion = (level, params) =>{
   // recursion terminator
   if(level > MAX_LEVEL){
     process_result
     return
   }
   // process current level
   process(level, params)
   //drill down
   recursion(level+1, params)
   //clean current level status if needed

}

File: geekbang/algorithm-models/3.3-递归代码模板/recursion.go
package main

// Go
func Recursion(level int, param int) {
	// terminator
	if level > MAX_LEVEL {
		return
	}
	// proccess current level
	Process(level, param)
	// drill down
	Recursion(level+1, param)
	// clean current level status if needed
}
